AdSecCoreTests/SlsResultsFunctionTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿using AdSecCore;
+using AdSecCore.Builders;
+using AdSecCore.Functions;
+
+using Oasys.AdSec;
+
+using OasysUnits;
+using OasysUnits.Units;
+
+namespace AdSecCoreTests.Functions {
+  public class SlsResultFunctionTest {
+    private readonly SlsResultFunction _component;
+    private static SectionSolution? Solution { get; set; } = null;
+    public SlsResultFunctionTest() {
+      _component = new SlsResultFunction();
+      if (Solution == null) {
+        Solution = new SolutionBuilder().Build();
+      }
+      _component.SolutionInput.Value = Solution;
+      _component.LoadInput.Value = ILoad.Create(Force.FromKilonewtons(100), Moment.FromKilonewtonMeters(100), Moment.Zero);
+    }
+
+    [Fact]
+    public void ShouldHaveCorrectMetadata() {
+      Assert.Equal("Find Crack Load", _component.Metadata.Name);
+      Assert.Equal("CrackLd", _component.Metadata.NickName);
+      Assert.Equal("Increases the load until set crack width is reached", _component.Metadata.Description);
+    }
+
+
+    [Fact]
+    public void ShouldHaveCorrectInputAttributes() {
+      var inputs = _component.GetAllInputAttributes();
+      Assert.Equal(2, inputs.Length);
+      Assert.IsType<SectionSolutionParameter>(inputs[0]);
+      Assert.IsType<GenericParameter>(inputs[1]);
+    }
+
+    [Fact]
+    public void ShouldHaveCorrectOutputAttributes() {
+      var outputs = _component.GetAllOutputAttributes();
+      Assert.Equal(7, outputs.Length);
+      Assert.IsType<LoadParameter>(outputs[0]);
+      Assert.IsType<CrackArrayParameter>(outputs[1]);
+      Assert.IsType<CrackParameter>(outputs[2]);
+      Assert.IsType<DoubleParameter>(outputs[3]);
+      Assert.IsType<DeformationParameter>(outputs[4]);
+      Assert.IsType<SecantStiffnessParameter>(outputs[5]);
+      Assert.IsType<IntervalArrayParameter>(outputs[6]);
+    }
+
+    [Fact]
+    public void ShouldRefreshDeformationDescription() {
+      _component.RefreshDeformation(StrainUnit.Ratio, CurvatureUnit.PerMeter);
+      Assert.Contains("Strain [ε]", _component.DeformationOutput.Description);
+      Assert.Contains("Curvature around zz (so in local y-direction) [m]", _component.DeformationOutput.Description);
+      Assert.Contains("Curvature around yy (so in local z-direction) [m]", _component.DeformationOutput.Description);
+    }
+
+    [Fact]
+    public void ShouldRefreshSecantStiffnessDescription() {
+      _component.RefreshSecantStiffness(AxialStiffnessUnit.Newton, BendingStiffnessUnit.NewtonSquareMeter);
+      Assert.Contains("Axial stiffness [N]", _component.SecantStiffnessOutput.Description);
+      Assert.Contains("The bending stiffness about the y-axis in the local coordinate system [Nm]", _component.SecantStiffnessOutput.Description);
+      Assert.Contains("The bending stiffness about the z-axis in the local coordinate system [Nm]", _component.SecantStiffnessOutput.Description);
```

Comment:
Maybe too much? The values of the string or the order is not critical to be tested? Let's discuss.